### ISSUE TITLE
i586-only test failures on openSUSE

### ISSUE THREAD

**Issue by jayvdb** [itamarst/crochet#125]
I'm re-packaging crochet for openSUSE, and encountering the following errors only on i586, on py2 and py3

```
[   70s] ======================================================================
[   70s] ERROR: test_later_failure_result (crochet.tests.test_api.EventualResultTests)
[   70s] test_later_failure_result
[   70s] ----------------------------------------------------------------------
[   70s] RuntimeError
[   70s] 
[   70s] ======================================================================
[   70s] FAIL: test_control_c_is_possible (crochet.tests.test_api.EventualResultTests)
[   70s] test_control_c_is_possible
[   70s] ----------------------------------------------------------------------
[   70s] Traceback (most recent call last):
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/internet/defer.py", line 151, in maybeDeferred
[   70s]     result = f(*args, **kw)
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/internet/utils.py", line 221, in runWithWarningsSuppressed
[   70s]     reraise(exc_info[1], exc_info[2])
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/python/compat.py", line 464, in reraise
[   70s]     raise exception.with_traceback(traceback)
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/internet/utils.py", line 217, in runWithWarningsSuppressed
[   70s]     result = f(*a, **kw)
[   70s]   File "/home/abuild/rpmbuild/BUILD/crochet-1.11.0/crochet/tests/test_api.py", line 384, in test_control_c_is_possible
[   70s]     self.assertEqual(process.wait(), 23)
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/trial/_synctest.py", line 434, in assertEqual
[   70s]     super(_Assertions, self).assertEqual(first, second, msg)
[   70s]   File "/usr/lib/python3.7/unittest/case.py", line 839, in assertEqual
[   70s]     assertion_func(first, second, msg=msg)
[   70s]   File "/usr/lib/python3.7/unittest/case.py", line 832, in _baseAssertEqual
[   70s]     raise self.failureException(msg)
[   70s] twisted.trial.unittest.FailTest: 1 != 23
[   70s] 
[   70s] ======================================================================
[   70s] FAIL: test_later_failure_result (crochet.tests.test_api.EventualResultTests)
[   70s] test_later_failure_result
[   70s] ----------------------------------------------------------------------
[   70s] Traceback (most recent call last):
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/internet/defer.py", line 151, in maybeDeferred
[   70s]     result = f(*args, **kw)
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/internet/utils.py", line 221, in runWithWarningsSuppressed
[   70s]     reraise(exc_info[1], exc_info[2])
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/python/compat.py", line 464, in reraise
[   70s]     raise exception.with_traceback(traceback)
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/internet/utils.py", line 217, in runWithWarningsSuppressed
[   70s]     result = f(*a, **kw)
[   70s]   File "/home/abuild/rpmbuild/BUILD/crochet-1.11.0/crochet/tests/test_api.py", line 192, in test_later_failure_result
[   70s]     (result_list[0], result_list[1].__class__), (False, RuntimeError))
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/trial/_synctest.py", line 434, in assertEqual
[   70s]     super(_Assertions, self).assertEqual(first, second, msg)
[   70s]   File "/usr/lib/python3.7/unittest/case.py", line 839, in assertEqual
[   70s]     assertion_func(first, second, msg=msg)
[   70s]   File "/usr/lib/python3.7/unittest/case.py", line 1056, in assertTupleEqual
[   70s]     self.assertSequenceEqual(tuple1, tuple2, msg, seq_type=tuple)
[   70s]   File "/usr/lib/python3.7/unittest/case.py", line 1027, in assertSequenceEqual
[   70s]     self.fail(msg)
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/trial/_synctest.py", line 377, in fail
[   70s]     raise self.failureException(msg)
[   70s] twisted.trial.unittest.FailTest: Tuples differ: (False, <class 'crochet._eventloop.TimeoutError'>) != (False, <class 'RuntimeError'>)
[   70s] 
[   70s] First differing element 1:
[   70s] <class 'crochet._eventloop.TimeoutError'>
[   70s] <class 'RuntimeError'>
[   70s] 
[   70s] - (False, <class 'crochet._eventloop.TimeoutError'>)
[   70s] + (False, <class 'RuntimeError'>)
[   70s] 
[   70s] ======================================================================
[   70s] FAIL: test_later_success_result (crochet.tests.test_api.EventualResultTests)
[   70s] test_later_success_result
[   70s] ----------------------------------------------------------------------
[   70s] Traceback (most recent call last):
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/internet/defer.py", line 151, in maybeDeferred
[   70s]     result = f(*args, **kw)
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/internet/utils.py", line 221, in runWithWarningsSuppressed
[   70s]     reraise(exc_info[1], exc_info[2])
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/python/compat.py", line 464, in reraise
[   70s]     raise exception.with_traceback(traceback)
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/internet/utils.py", line 217, in runWithWarningsSuppressed
[   70s]     result = f(*a, **kw)
[   70s]   File "/home/abuild/rpmbuild/BUILD/crochet-1.11.0/crochet/tests/test_api.py", line 162, in test_later_success_result
[   70s]     self.assertEqual(result_list, [True, 345])
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/trial/_synctest.py", line 434, in assertEqual
[   70s]     super(_Assertions, self).assertEqual(first, second, msg)
[   70s]   File "/usr/lib/python3.7/unittest/case.py", line 839, in assertEqual
[   70s]     assertion_func(first, second, msg=msg)
[   70s]   File "/usr/lib/python3.7/unittest/case.py", line 1045, in assertListEqual
[   70s]     self.assertSequenceEqual(list1, list2, msg, seq_type=list)
[   70s]   File "/usr/lib/python3.7/unittest/case.py", line 1027, in assertSequenceEqual
[   70s]     self.fail(msg)
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/trial/_synctest.py", line 377, in fail
[   70s]     raise self.failureException(msg)
[   70s] twisted.trial.unittest.FailTest: Lists differ: [False, TimeoutError()] != [True, 345]
[   70s] 
[   70s] First differing element 0:
[   70s] False
[   70s] True
[   70s] 
[   70s] - [False, TimeoutError()]
[   70s] + [True, 345]
[   70s] 
[   70s] ======================================================================
[   70s] FAIL: test_control_c_is_possible (crochet.tests.test_api.WaitForReactorTests)
[   70s] test_control_c_is_possible
[   70s] ----------------------------------------------------------------------
[   70s] Traceback (most recent call last):
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/internet/defer.py", line 151, in maybeDeferred
[   70s]     result = f(*args, **kw)
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/internet/utils.py", line 221, in runWithWarningsSuppressed
[   70s]     reraise(exc_info[1], exc_info[2])
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/python/compat.py", line 464, in reraise
[   70s]     raise exception.with_traceback(traceback)
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/internet/utils.py", line 217, in runWithWarningsSuppressed
[   70s]     result = f(*a, **kw)
[   70s]   File "/home/abuild/rpmbuild/BUILD/crochet-1.11.0/crochet/tests/test_api.py", line 1057, in test_control_c_is_possible
[   70s]     self.assertEqual(process.wait(), 23)
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/trial/_synctest.py", line 434, in assertEqual
[   70s]     super(_Assertions, self).assertEqual(first, second, msg)
[   70s]   File "/usr/lib/python3.7/unittest/case.py", line 839, in assertEqual
[   70s]     assertion_func(first, second, msg=msg)
[   70s]   File "/usr/lib/python3.7/unittest/case.py", line 832, in _baseAssertEqual
[   70s]     raise self.failureException(msg)
[   70s] twisted.trial.unittest.FailTest: 1 != 23
[   70s] 
[   70s] ======================================================================
[   70s] FAIL: test_reactor_stop_unblocks (crochet.tests.test_api.WaitForReactorTests)
[   70s] test_reactor_stop_unblocks
[   70s] ----------------------------------------------------------------------
[   70s] Traceback (most recent call last):
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/internet/defer.py", line 151, in maybeDeferred
[   70s]     result = f(*args, **kw)
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/internet/utils.py", line 221, in runWithWarningsSuppressed
[   70s]     reraise(exc_info[1], exc_info[2])
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/python/compat.py", line 464, in reraise
[   70s]     raise exception.with_traceback(traceback)
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/internet/utils.py", line 217, in runWithWarningsSuppressed
[   70s]     result = f(*a, **kw)
[   70s]   File "/home/abuild/rpmbuild/BUILD/crochet-1.11.0/crochet/tests/test_api.py", line 1085, in test_reactor_stop_unblocks
[   70s]     self.assertEqual(process.wait(), 23)
[   70s]   File "/usr/lib/python3.7/site-packages/twisted/trial/_synctest.py", line 434, in assertEqual
[   70s]     super(_Assertions, self).assertEqual(first, second, msg)
[   70s]   File "/usr/lib/python3.7/unittest/case.py", line 839, in assertEqual
[   70s]     assertion_func(first, second, msg=msg)
[   70s]   File "/usr/lib/python3.7/unittest/case.py", line 832, in _baseAssertEqual
[   70s]     raise self.failureException(msg)
[   70s] twisted.trial.unittest.FailTest: 1 != 23
[   70s] 
```

I also encounter failure in `test_timeout` on zSystems s390x intermittently, last seen on py37, but may also be occurring on py27
```
[   88s] ======================================================================
[   88s] FAIL: test_timeout (crochet.tests.test_api.EventualResultTests)
[   88s] test_timeout
[   88s] ----------------------------------------------------------------------
[   88s] Traceback (most recent call last):
[   88s]   File "/usr/lib64/python3.7/site-packages/twisted/internet/defer.py", line 151, in maybeDeferred
[   88s]     result = f(*args, **kw)
[   88s]   File "/usr/lib64/python3.7/site-packages/twisted/internet/utils.py", line 221, in runWithWarningsSuppressed
[   88s]     reraise(exc_info[1], exc_info[2])
[   88s]   File "/usr/lib64/python3.7/site-packages/twisted/python/compat.py", line 464, in reraise
[   88s]     raise exception.with_traceback(traceback)
[   88s]   File "/usr/lib64/python3.7/site-packages/twisted/internet/utils.py", line 217, in runWithWarningsSuppressed
[   88s]     result = f(*a, **kw)
[   88s]   File "/home/abuild/rpmbuild/BUILD/crochet-1.11.0/crochet/tests/test_api.py", line 209, in test_timeout
[   88s]     self.assertTrue(abs(time.time() - start - 0.03) < 0.005)
[   88s]   File "/usr/lib64/python3.7/site-packages/twisted/trial/_synctest.py", line 397, in assertTrue
[   88s]     super(_Assertions, self).assertTrue(condition, msg)
[   88s]   File "/usr/lib64/python3.7/unittest/case.py", line 692, in assertTrue
[   88s]     raise self.failureException(msg)
[   88s] twisted.trial.unittest.FailTest: False is not true
```

Package at https://build.opensuse.org/package/show/home:jayvdb:py-submit/python-crochet in case anyone wants to create a branch and debug it on OBS.

**Comment by itamarst** [comment]
Hi,

Thanks for doing the packaging, and for reporting the bug. Having 32-bit-only errors is rather fascinating, given what Crochet does. I'll see if anything leaps out at me.

**Comment by itamarst** [comment]
I believe this should be fixed in the newly released Crochet 1.12.

**Comment by jayvdb** [comment]
Hi @itamarst , some of the failures are fixed, but I am still seeing two

```py
[   45s] ======================================================================
[   45s] FAIL: test_control_c_is_possible (crochet.tests.test_api.WaitForReactorTests)
[   45s] A call to a decorated function responds to a Ctrl-C (i.e. with a
[   45s] ----------------------------------------------------------------------
[   45s] Traceback (most recent call last):
[   45s]   File "/usr/lib/python3.8/site-packages/twisted/internet/defer.py", line 151, in maybeDeferred
[   45s]     result = f(*args, **kw)
[   45s]   File "/usr/lib/python3.8/site-packages/twisted/internet/utils.py", line 221, in runWithWarningsSuppressed
[   45s]     reraise(exc_info[1], exc_info[2])
[   45s]   File "/usr/lib/python3.8/site-packages/twisted/python/compat.py", line 464, in reraise
[   45s]     raise exception.with_traceback(traceback)
[   45s]   File "/usr/lib/python3.8/site-packages/twisted/internet/utils.py", line 217, in runWithWarningsSuppressed
[   45s]     result = f(*a, **kw)
[   45s]   File "/home/abuild/rpmbuild/BUILD/crochet-1.12.0/crochet/tests/test_api.py", line 1057, in test_control_c_is_possible
[   45s]     self.assertEqual(process.wait(), 23)
[   45s]   File "/usr/lib/python3.8/site-packages/twisted/trial/_synctest.py", line 434, in assertEqual
[   45s]     super(_Assertions, self).assertEqual(first, second, msg)
[   45s]   File "/usr/lib/python3.8/unittest/case.py", line 912, in assertEqual
[   45s]     assertion_func(first, second, msg=msg)
[   45s]   File "/usr/lib/python3.8/unittest/case.py", line 905, in _baseAssertEqual
[   45s]     raise self.failureException(msg)
[   45s] twisted.trial.unittest.FailTest: 1 != 23
[   45s] 
[   45s] ======================================================================
[   45s] FAIL: test_reactor_stop_unblocks (crochet.tests.test_api.WaitForReactorTests)
[   45s] Any @wait_for_reactor-decorated calls still waiting when the reactor
[   45s] ----------------------------------------------------------------------
[   45s] Traceback (most recent call last):
[   45s]   File "/usr/lib/python3.8/site-packages/twisted/internet/defer.py", line 151, in maybeDeferred
[   45s]     result = f(*args, **kw)
[   45s]   File "/usr/lib/python3.8/site-packages/twisted/internet/utils.py", line 221, in runWithWarningsSuppressed
[   45s]     reraise(exc_info[1], exc_info[2])
[   45s]   File "/usr/lib/python3.8/site-packages/twisted/python/compat.py", line 464, in reraise
[   45s]     raise exception.with_traceback(traceback)
[   45s]   File "/usr/lib/python3.8/site-packages/twisted/internet/utils.py", line 217, in runWithWarningsSuppressed
[   45s]     result = f(*a, **kw)
[   45s]   File "/home/abuild/rpmbuild/BUILD/crochet-1.12.0/crochet/tests/test_api.py", line 1085, in test_reactor_stop_unblocks
[   45s]     self.assertEqual(process.wait(), 23)
[   45s]   File "/usr/lib/python3.8/site-packages/twisted/trial/_synctest.py", line 434, in assertEqual
[   45s]     super(_Assertions, self).assertEqual(first, second, msg)
[   45s]   File "/usr/lib/python3.8/unittest/case.py", line 912, in assertEqual
[   45s]     assertion_func(first, second, msg=msg)
[   45s]   File "/usr/lib/python3.8/unittest/case.py", line 905, in _baseAssertEqual
[   45s]     raise self.failureException(msg)
[   45s] twisted.trial.unittest.FailTest: 1 != 23
```
